spot_driver/spot_driver/spot_ros2.py

```diff
@@ -445,87 +456,128 @@ def _goal_complete(self, feedback):
+            # So if any one of the sub-commands is still in progress, it short-circuits out as
+            # IN_PROGRESS.  And if it makes it to the bottom of the function, then all components
+            # must be satisfied, and it returns SUCCESS.
+            # One corner case to know about is that the commands that don't provide feedback, such
+            # as a velocity command will return SUCCESS.  This allows you to use them more effectively
+            # with other commands.  For example if you wanted to move the arm with some velocity
+            # while the mobility is going to some SE2 trajectory then that will work.
+
+            sync_feedback = feedback.command.synchronized_feedback
+            if sync_feedback.arm_command_feedback_is_set == True:
+                arm_feedback =  sync_feedback.arm_command_feedback
+                if (arm_feedback.status.value != arm_feedback.status.STATUS_PROCESSING):
+                    return IN_PROGRESS
+                if (arm_feedback.feedback.feedback_choice == arm_feedback.feedback.FEEDBACK_ARM_CARTESIAN_FEEDBACK_SET):
+                    if (arm_feedback.feedback.arm_cartesian_feedback.status.value != arm_feedback.feedback.arm_cartesian_feedback.status.STATUS_TRAJECTORY_COMPLETE):
+                        return IN_PROGRESS
+                elif (arm_feedback.feedback.feedback_choice == arm_feedback.feedback.FEEDBACK_ARM_JOINT_MOVE_FEEDBACK_SET):
+                    if (arm_feedback.feedback.arm_joint_move_feedback.status.value != arm_feedback.feedback.arm_joint_move_feedback.status.STATUS_COMPLETE):
+                        return IN_PROGRESS
+                elif (arm_feedback.feedback.feedback_choice == arm_feedback.feedback.FEEDBACK_NAMED_ARM_POSITION_FEEDBACK_SET):
+                    if (arm_feedback.feedback.named_arm_position_feedback.status.value != arm_feedback.feedback.named_arm_position_feedback.STATUS_COMPLETE):
+                        return IN_PROGRESS
+                elif (arm_feedback.feedback.feedback_choice == arm_feedback.feedback.FEEDBACK_ARM_VELOCITY_FEEDBACK_SET):
+                    # ArmVelocityCommand provides no feedback
+                    print(COLOR_YELLOW + 'WARNING: ArmVelocityCommand provides no feedback' + COLOR_END)
+                    pass
+                elif (arm_feedback.feedback.feedback_choice == arm_feedback.feedback.FEEDBACK_ARM_GAZE_FEEDBACK_SET):
+                    if (arm_feedback.feedback.arm_gaze_feedback.status.value != arm_feedback.feedback.arm_gaze_feedback.STATUS_TRAJECTORY_COMPLETE):
+                        return IN_PROGRESS
+                elif (arm_feedback.feedback.feedback_choice == arm_feedback.feedback.FEEDBACK_ARM_STOP_FEEDBACK_SET):
+                    # Stop command provides no feedback
+                    print(COLOR_YELLOW + 'WARNING: Stop command provides no feedback' + COLOR_END)
+                    pass
```

Comment:
this should return SUCCESS i think

Reply:
What if you want to stop the arm while simultaneously moving mobility somewhere?  `pass`ing here means that you can still be running a mobility (and maybe gripper) command too.  and if not, then it will succeed at the bottom of the function.  

this is the logic i was trying to explain at the top of the function.  but probably could have been better.

Reply:
Oops sorry I meant to delete this comment.